tests/governance/test-question-batch-gate.sh

```
@@ -0,0 +1,183 @@
+O="$(jq -cn '{session_id:"h",prompt_id:"h2",last_assistant_message:"a?\nb?\nc?"}' \
+  | env -u HOME -u QBG_STATE_DIR PATH="$PATH" bash "$HOOK" 2>&1)"; RC=$?
+[ "$RC" -eq 0 ] && ok "unset HOME -> exit 0 (BLOCKING-1)" \
+  || bad "unset HOME" "rc=$RC out=$(printf '%s' "$O" | head -1)"
+
+# BLOCKING-2: payload-controlled ids must never escape STATE_DIR
+VICTIM="$TMP/victim"; mkdir -p "$VICTIM"
+SID="../../victim/pwned" emit "p9" "a?
+b?
+c?"
+[ ! -e "$VICTIM/1.pwned.2.p9.marker.d" ] && [ "$RC" -eq 0 ] \
+  && ok "traversal via session_id blocked (BLOCKING-2)" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)